lib/companies/pending_changes.ex

```diff
@@ -47,62 +30,66 @@ defmodule Companies.PendingChanges do
     end
   end
 
-  defp apply_changes("create", changeset), do: Repo.insert(changeset)
-  defp apply_changes("update", changeset), do: Repo.update(changeset)
+  def create(changeset, action, user, note \\ "")
+  def create(%{valid?: false} = changes, action, _user, _note), do: invalid_change(changes, action)
+  def create(changes, action, user, note), do: insert_change(changes, action, note, user)
 
-  defp changeset(module, changes) do
-    record =
-      if Map.has_key?(changes, "id") do
-        id = Map.get(changes, "id")
-        Repo.get(module, id)
-      else
-        struct(module, %{})
-      end
+  def act(change_id, "false") do
+    PendingChange
+    |> Repo.get!(change_id)
+    |> Repo.delete!()
```

Comment:
We're not deleting anything. We just mark it as rejected. We want the record.

Reply:
Oh ok, nice. I will fix this.